Sources/Moya/MultiTarget.swift

```diff
@@ -21,9 +21,10 @@ public enum MultiTarget: TargetType {
         return target.method
     }
 
-    public var sampleData: Data {
-        return target.sampleData
-    }
+// TODO: - Not sure how this would affect sample data
+//    public var sampleData: Data {
+//        return target.sampleData
+//    }
```

Comment:
This is also the only thing I haven't been able to get around with this PR. I don't use the `MultiTarget` so I don't really understand it's purpose.

Reply:
Yeah, this is tricky. I will think about this one, but two ideas that come to my mind right now are:
1. Create two MultiTargets, one that stubs and one that does not (like with target types)
2. If target is not a Testing one, use empty data/crash, but this is not the best solution I think.

Reply:
This PR is very low priority IMO. It won't make it in Moya 12.0 for sure and I'd like a few Moya members to discuss if they even like this change.

Reply:
Revisited this today... I have a feeling it "just works" now that I understand what `MultiTarget` is doing.
If the multi-target type is a `TestTargetType` the user will have to specify where to get `sampleData` from.

Following the pattern of the existing properties, they would have to extract the associated value of the `target` case. They would have to cast to `TestTargetType` to access the underlying `sampleData`. If casting fails they still have to return something because `sampleData` is non-optional. Really hard to go unnoticed unless you're already writing bad code to begin with.


So as the other properties are doing, they would unwrap the associated value of the enum and return the data from the nested target.

Reply:
@SD10 I'm not sure I follow... Are you saying that we won't provide any support for `MultiTarget` with `TestTargetType` so the users would have to do it themselves?

Reply:
@sunshinejr I wouldn't say "we won't provide any support" but I was going to suggest that they extend `MultiTarget` to conform to `TestTargetType` inside their test target. We could add a `MultiTestTarget` if you believe this is expecting too much.

Reply:
@SD10 Oh, sorry if it sounded like that, I was just trying to understand if that's what you would do. I don't think there is a 100% good solution for this, `MultiTestTarget` is okay, but still doesn't cover all cases (like mixed targets & test targets). I wonder if making few examples instead would be enough... What do you think?

Reply:
`MultiTestTarget` would cover all cases because the enum associated value would have to be `TestTargetType` as well. 

An edge case would be a `MultiTarget` with an inner `TestTargetType`. Stubbing would be determined by the type of the root `MultiTarget` so it would never stub.

We could document this. Even provide an assertion/error if the inner target is a `TestTargetType` but `MultiTarget` is not? The only way to fully prevent this is to check every target at runtime by trying to cast it to `TestTargetType`.

Reply:
Honestly, the edge case allows you to opt in or out of testing by wrapping your `TargetType` or `TestTargetType`. It's valuable behavior in a way 😂 

Let's sit on this a bit more for me to explore casting. To be honest, I really liked the code separation that the `where` clause provides, but if I can protect people from their own design decisions then that should be explored.

Reply:
Realized last night in a dream that nothing can be done with casting. The `MultiTarget` abstracts away any sense of an underlying `TargetType`.